apps/verticalhome/js/app.js

```diff
@@ -158,6 +187,182 @@
 
     },
 
+    saveAllPinnedApps: function(){
+      var pinnedItems= app.pinnedAppsManager.items;
+      if(this.deletedApps.length>0){
+        for(var x = 0; x < pinnedItems.length; x++){
+          if(pinnedItems[x].element.style.display == 'none'){
+            pinnedItems.splice(x,1);
+          }
+        }
+
+        for(var z = 0; z < this.deletedApps.length; z++){
+          app.itemStore.deletePinnedAppItem(this.deletedApps[z]);
+        }
+      }
+
+      for(var i = 0; i<pinnedItems.length; i++){
+        var obj = {};
+
+        if(pinnedItems[i].entry!=null){
+
+          obj.name = pinnedItems[i].entry.name;
+          obj.manifestURL = typeof(
+            pinnedItems[i].entry.manifestURL
+            ) == 'undefined'?
+            pinnedItems[i].entry.targetApp.manifestURL
+            : pinnedItems[i].entry.manifestURL;
```

Comment:
I can not understand it fully, but it seems there is a fault or unnecessary code.

At first you can pinnedItems[i].entry move to separate variable and this block would be more understandable.
Next why are you check type of the variable on undefined.

I think it would be enough if you write something like this:
var curEntry = pinnedItems[i].entry;
obj.manifestURL = curEntry.manifestURL || curEntry.targetApp.manifestURL;